File: 010_summation_of_primes/010.rb
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
#
# Find the sum of all the primes below two million.

require 'prime'

def sum_of_primes(upper_bound)
  count = 0
  sum = 0

  while count < upper_bound
    sum += count if count.prime?
    count += 1
  end

  sum
end

p sum_of_primes(2000000)

File: 010_summation_of_primes/010.js
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//
// Find the sum of all the primes below two million.

const sumOfPrimes = upperBound => {
  let primeArr = [2];
  let primeSum = 2;

  for (let i = 3; i >= 0; i++) {
    let prime = true;

    primeArr.forEach(num => {
      if (i % num === 0) {
        prime = false;
      }
    });

    if (prime) {
      primeSum += i;
      primeArr.push(i);
    }

    if (i === upperBound) return primeSum;
  }
};

console.log(sumOfPrimes(2000000));
